packages/perspective-viewer-d3fc/test/js/integration/events.spec.js

```diff
@@ -0,0 +1,146 @@
+    return await page.evaluate(async () => {
+        const viewer = document
+            .querySelector(`perspective-viewer perspective-viewer-d3fc-yline`)
+            .shadowRoot.querySelector("svg");
+        return viewer.outerHTML || "MISSING";
+    });
+}
+
+utils.with_server({}, () => {
+    describe.page(
+        "events.html",
+        () => {
+            test.capture(
+                "perspective-config-update event is fired when series axis is changed",
+                async (page) => {
+                    // Await the viewer element to exist on the page
+                    const viewer = await page.$("perspective-viewer");
+                    await page.evaluate(async (viewer) => {
+                        // Await the table load
+                        await viewer.getTable();
+
+                        await viewer.restore({
+                            plugin: "Y Line",
+                            columns: ["Sales", "Profit"],
+                        });
+
+                        // Register a listener for `perspective-config-update` event
+                        window.__series_events__ = [];
+                        viewer.addEventListener(
+                            "perspective-config-update",
+                            (evt) => {
+                                window.__series_events__.push(evt);
+                            }
+                        );
+                    }, viewer);
+                    const axisLabel = (
+                        await page.waitForFunction(() =>
+                            document
+                                .querySelector("perspective-viewer-d3fc-yline")
+                                .shadowRoot.querySelector(
+                                    ".y-label .splitter-label"
+                                )
+                        )
+                    ).asElement();
+                    await axisLabel.click(axisLabel);
+
+                    const count = await page.evaluate(async (viewer) => {
+                        // Await the plugin rendering
+                        await viewer.flush();
+
+                        // Count the events;
+                        return window.__series_events__.length;
+                    }, viewer);
+
+                    // Expect 1 event
+                    expect(count).toEqual(1);
+
+                    // Return the chart contents
+                    return get_contents(page);
+                }
+            );
+
+            test.capture(
+                "perspective-config-update event is fired when legend position is changed",
+                async (page) => {
+                    // Await the viewer element to exist on the page
+                    const viewer = await page.$("perspective-viewer");
+                    await page.evaluate(async (viewer) => {
+                        // Await the table load
+                        await viewer.getTable();
+
+                        viewer.restore({
```

Comment:
```suggestion
                        await viewer.restore({
```
Good catch on the test above. Probably want to do the same here.

Reply:
fixed

Reply:
Actually this was more nefarious than it appears - turns out that this test had a similar issue, `mouse.down()` was called to initiate the drag in the test, but not `mouse.up()` which fires the drag end event.  The test "worked" by asserting the `perspective-config-update` which was fired by `restore()`.